### ISSUE TITLE
LGTM.com - false positive (Function <F> should return a value of type enable_if_t but does not return a value here)

### ISSUE THREAD

**Issue by oknenavin** [github/codeql#7248]
This is something new, I'm sure it was OK a couple of days before.
The type of enable_if_t is the type of its second template argument.

https://lgtm.com/projects/g/oknenavin/cxon/snapshot/6006a81089265b2cf034ef73c8111bd00564700b/files/src/cxon/lang/common/cio/class.hxx?sort=name&dir=ASC&mode=heatmap#x92141d088169c5cf:1

**Comment by oknenavin** [comment]
C++

**Comment by aeisenberg** [comment]
Indeed, this looks like a false positive. Thank you for reporting it!

Our current focus is on improving our security analysis. Normally, we will put non-security related FP requests on the backlog, but since you are saying this is a regression, I'll ask our C++ team to take a look at this.

GitHub Code Scanning and lgtm.com have facilities for suppressing individual alerts or disabling a query.

**Comment by oknenavin** [comment]
OK, thanks.
